planning/behavior_velocity_planner/include/scene_module/occlusion_spot/occlusion_spot_utils.hpp

```diff
@@ -118,53 +142,98 @@ struct PossibleCollisionInfo
   }
 };
 
+lanelet::ConstLanelet toPathLanelet(const PathWithLaneId & path);
+
+// Note : consider offset_from_start_to_ego and safety margin for collision here
+inline void handleCollisionOffset(
+  std::vector<PossibleCollisionInfo> & possible_collisions, double offset, double margin)
+{
+  for (auto & pc : possible_collisions) {
+    pc.arc_lane_dist_at_collision.length -= offset;
+    pc.arc_lane_dist_at_collision.length -= margin;
+  }
+}
+
+inline double offsetFromStartToEgo(
+  const PathWithLaneId & path, const Pose & ego_pose, const int closest_idx)
+{
+  double offset_from_ego_to_closest = 0;
+  for (int i = 0; i < closest_idx; i++) {
+    const auto & curr_p = path.points[i].point.pose.position;
+    const auto & next_p = path.points[i + 1].point.pose.position;
+    offset_from_ego_to_closest += tier4_autoware_utils::calcDistance2d(curr_p, next_p);
+  }
+  const double offset_from_closest_to_target =
+    -planning_utils::transformRelCoordinate2D(ego_pose, path.points[closest_idx].point.pose)
+       .position.x;
+  return offset_from_ego_to_closest + offset_from_closest_to_target;
+}
+
+inline void clipPathByLength(
+  const PathWithLaneId & path, PathWithLaneId & clipped, const double max_length = 100.0)
+{
+  double length_sum = 0;
+  for (int i = 0; i < static_cast<int>(path.points.size()) - 1; i++) {
+    length_sum += tier4_autoware_utils::calcDistance2d(path.points.at(i), path.points.at(i + 1));
+    if (length_sum > max_length) return;
+    clipped.points.emplace_back(path.points.at(i));
+  }
+}
+
+inline bool isStuckVehicle(PredictedObject obj, const double min_vel)
+{
+  if (
+    obj.classification.at(0).label == ObjectClassification::CAR ||
+    obj.classification.at(0).label == ObjectClassification::TRUCK ||
+    obj.classification.at(0).label == ObjectClassification::BUS) {
+    if (obj.kinematics.initial_twist_with_covariance.twist.linear.x < min_vel) {
```

Comment:
std::abs?

Reply:
maybe it's better for moving vehicle backward.

Reply:
fixed at 0f70107